### PR TITLE
Add PropertyMapping and corresponding tests

### PR DESCRIPTION
**Pull request by BentolhodaH**
Add PropertyMapping and corresponding tests

### CODE REVIEW

File: ExampleLPG2RDF.ttl [hunk 1]
```diff
@@ -30,6 +30,12 @@ PREFIX ex:     <http://example.org/>
 #ex:IRIBasedEdgeLabelMapping a rdfs:Class ;
 #   rdfs:subClassOf ex:EdgeLabelMapping .
 
+#ex:PropertyMapping a rdfs:Class ;
+#
+#ex:IRIBasedPropertyMapping a rdfs:Class ;
+#   rdfs:subClassOf ex:PropertyMapping .
```
Comment (hartig) [inline ExampleLPG2RDF.ttl:36]: Can you please call these classes `PropertyNameMapping` and `IRIBasedPropertyNameMapping` in order to be aligned with the terminology used in the formalization of LPG-to-RDF configurations.

Similarly, the property should be renamed from `propertyMapping` to `propertyNameMapping`.

And, the naming in the Java code should be adapted accordingly (e.g., the name of the interface, of the classes, of the member variables).

File: src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/impl/LPG2RDFConfigurationImpl.java [hunk 2]
```diff
@@ -7,21 +7,20 @@
 
 public class LPG2RDFConfigurationImpl implements LPG2RDFConfiguration {
 
-    protected final String NS = "https://example.org/";
-    protected final String PROPERTY = "property/";
-
     protected final Node label;
     protected final NodeMapping nodeMapping;
```
Comment (hartig) [inline src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/impl/LPG2RDFConfigurationImpl.java:12]:
```suggestion
```

File: src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/impl/PropertyMapping.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
+package se.liu.ida.hefquin.engine.wrappers.lpgwrapper.impl;
+
+import org.apache.jena.graph.Node;
+
+
+/**
+ * This interface captures the notion of a property name mapping that is part of the notion of an LPG-to-RDF configuration,
+ * and that such a property mapping is an injective function from properties in LPGs to IRIs nodes as
```
Comment (hartig) [inline src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/impl/PropertyMapping.java:8]:
```suggestion
 * where such a property name mapping is an injective function from names of properties in LPGs to IRIs as
```

File: src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/impl/PropertyMapping.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+ * can occur in RDF graphs.
+ * This interface contains the functions:
+ *  -map: map String of properties to IRIs properties
+ *  -unmap: map IRIs properties to String
+ *  -isPossibleResult: Check if the given RDF term is in the image of this property mapping
+ */
+public interface PropertyMapping {
+    /**
+     * Returns a URI property (in the form of a Jena {@link Node} object) for the given String.
+     * It applies this property mapping to the given String.
```
Comment (hartig) [inline src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/impl/PropertyMapping.java:18]:
```suggestion
     * Returns the IRI (in the form of a Jena {@link Node} object) that is the result of
     * applying this property name mapping to the given property name.
```

File: src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/impl/LPG2RDFConfigurationImpl.java [hunk 5]
```diff
@@ -54,13 +53,13 @@ public String unmapEdgeLabel(final Node node) {
     }
 
     @Override
-    public Node mapProperty(final String property) {
-        return propertyMapping.map(property);
+    public Node mapProperty(final String propertyName) {
```
Comment (BentolhodaH) [inline src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/impl/LPG2RDFConfigurationImpl.java:56]: I changed propertyMapping to propertyNameMapping in other parts of the code, but I was unsure about three methods in this class(mapProperty, unmapProperty, and mapsToProperty). Is it necessary to change the names of these methods and their names in the superclass(LPG2RDFConfiguration) or not?

Comment (hartig) [reply src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/impl/LPG2RDFConfigurationImpl.java:56]: Thanks for asking! I think that these method names can remain as they are, for the moment at least.

**Pull request merged**